src/options.rs

```diff
@@ -848,8 +852,14 @@ where
 
     let output = if let Some(path) = matches.value_of("output") {
         let file = File::create(path)?;
+        if let Some(depfile) = matches.value_of("depfile") {
+            builder = builder.depfile(path, depfile);
+        }
         Box::new(io::BufWriter::new(file)) as Box<dyn io::Write>
     } else {
+        if matches.value_of("depfile").is_some() {
+            panic!("--depfile can only be provided if --output is.");
```

Comment:
Maybe elaborate a bit at least in a comment?

Or maybe we should follow `gcc` or `clang` and just output `-` instead of the filename? I could see it could be useful perhaps. For example, gcc:

```
$ gcc -M -E -x c++ -
#include "foo.h"
int main() {}
-: /usr/include/stdc-predef.h foo.h
```